akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -670,9 +670,12 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         flushingPromise.future
       }
     implicit val ec = remoteDispatcher
-    flushing.recover { case _ ⇒ Done }.map { _ ⇒
-      killSwitch.shutdown()
 
+    for {
+      _ ← flushing.recover { case _ ⇒ Done }
+      _ = killSwitch.shutdown()
+      _ ← streamsCompleted
```

Comment:
you can change the `attachStreamRestart` methods to do logging of errors also during shutdown, they are ignored there in `case _ if isShutdown`. Then we have all such logging in one place.